addons/web/static/src/views/list/list_renderer.js

```diff
@@ -1455,6 +1490,118 @@ export class ListRenderer extends Component {
+            const propertyColumn = {
+                ...baseFieldObject,
+                FieldComponent: PropertyValue,
+                rawAttrs: {
+                    ...baseFieldObject.rawAttrs,
+                    name: propertyName,
+                },
+                name: propertyName,
+                label: property.string,
+                propertyType: property.type,
+                string: property.string,
+            };
+            fields.push(propertyColumn)
+        }
+        return fields;
+    }
+
+    /**
+     * Create columns for properties
+     * @param {Array} columns all the columns (typically props.archInfo.columns)
+     * @returns the same array with the properties columns replaced by 1 column per property
+     */
+    propertiesGetAllColumns(columns) {
+        columns = [...columns];
+        for(const propertyName of this.getPropertiesFieldNames()) {
+            const propertiesColumnIndex = columns.findIndex(column => column.name === propertyName);
+            if(propertiesColumnIndex >= 0) {
+                const baseColumnObject = columns[propertiesColumnIndex];
+                const propertyColumns = this.getPropertyFieldsFromPropertiesField(baseColumnObject);
+                let property_column_id = 1;
+                for(const column of propertyColumns){
+                    column.id = `${column.id}_property_${property_column_id}`;
+                    property_column_id++;
+                }
+                columns.splice(propertiesColumnIndex, 1, ...propertyColumns);
+            }
+        }
+        return columns;
+    }
+
+    /**
+     * Replace 'properties' field with a field for each property value
+     * preserving most of the attributes of the original field.
+     * @param {Object} fields in the form of fieldInfo (e.g. this.props.list.fields)
+     * @returns input or altered copy of the input
+     */
+    propertiesProcessFields(fields) {
+        const propertiesFieldNames = this.getPropertiesFieldNames();
+        if(!propertiesFieldNames) {
+            return fields;
+        }
+        fields = {...fields};
+        for(const propertiesFieldName of propertiesFieldNames){
+            const properties = fields[propertiesFieldName];
+            if(properties) {
+                for(const property of this.getPropertyFieldsFromPropertiesField(properties)){
+                    fields[this.getPropertyFieldName(property.name, propertiesFieldName)] = property;
+                }
+                delete fields[propertiesFieldName];
+            }
+        }
+        return fields;
+    }
+
+    /**
+     * Turns the data value of the Properties fields into individual data values
+     * @param {Object} records
+     * @returns the same records Object with additional data values
+     */
+    propertiesProcessRecords(records) {
+        const propertiesFieldNames = this.getPropertiesFieldNames();
+        if(!propertiesFieldNames) {
+            return records;
+        }
```

Comment:
not useful imo (and it should be `propertiesFieldNames.length`, but better to remove it)